src/ClientData/include/ClientData/ProcessData.h

```diff
@@ -73,10 +73,17 @@ class ProcessData final {
 
   [[nodiscard]] ErrorMessageOr<ModuleInMemory> FindModuleByAddress(uint64_t absolute_address) const;
 
-  std::optional<uint64_t> GetModuleBaseAddress(const std::string& module_path) const;
-  absl::node_hash_map<std::string, ModuleInMemory> GetMemoryMapCopy() const;
+  // This function is deprecated since it relies on the fact that only one instance
+  // of module is loaded in the process which is not always true.
+  [[deprecated]] std::optional<uint64_t> GetModuleBaseAddress(const std::string& module_path) const;
 
-  [[nodiscard]] std::optional<ModuleInMemory> FindModuleByPath(
+  // This function is deprecated since it relies on the fact that only one instance
+  // of module is loaded in the process which is not always true.
+  [[deprecated]] absl::node_hash_map<std::string, ModuleInMemory> GetMemoryMapCopy() const;
+
+  // This function is deprecated since it relies on the fact that only one instance
+  // of module is loaded in the process which is not always true.
+  [[nodiscard, deprecated]] std::optional<ModuleInMemory> FindModuleByPath(
```

Comment:
I'm a bit skeptical about adding `[[deprecated]]` in this case, as it's not like there's an alternative to it at the time of this commit. It's not a problem if these are just going to be removed soon.

Reply:
we will have one peace of code that will be using it after fixing up other references and that is Deseraalization from the old file format. I think we should be able to move disassebly away from this (but it could also end up using it for some time).

I would like to avoid to have new code using it - hence [[depracated]]

Reply:
> that will be using it after fixing up other references

> I would like to avoid to have new code using it

Yes, I agree, this is what it's for. What I meant is that since you said that that's addressed in an upcoming change, I would have expected the `[[deprecated]]` to also be added in that change, instead of in advance (before the alternative is added).

Reply:
makes sense - removed deprecated attribute from this commit - will introduce in the follow-up.